### PR TITLE
MWPW-152581 Extension to promote to use upload report locked status in excel and ui

### PR DESCRIPTION
**Pull request by raga-adbe-gh**
Updates

1. Report as locked when the file is locked / could not be written due to file being open (in Copy or Promote)
2. In parallel waits for all request to be processed before picking next. Using the inparallel (from miloc) which will pick next tasks as and when a previous task completes.
3. Using parts of request wrapper (from miloc) for handling the status code and has option to skip status when required.
4. Timeout  (Status code 600) or Fetch errors (Status code 503) are also retry
5. Locked flag is passed over for reporting
6. Test cases updated

Tests - 
Scenario 1 - Copy from Main Tree to Floodgate Tree retain version
image

Scenario 2 - File is opened and copy is triggered
image

Scenario 3 - File is locked and copy is triggered
| Reported in GUI | Reported in Excel |
| -------- | -------- |
| image | image | 

Scenario 4 - File is opened (or Locked) and promote is triggered
| Reported in GUI | Reported in Excel |
| -------- | -------- |
| image | image |

### CODE REVIEW

File: actions/fgCopyActionHelper.js [hunk 1]
```diff
@@ -54,12 +52,13 @@ class FloodgateActionHelper {
                             // Save the file in the floodgate destination location
                             const saveStatus = await sharepoint.saveFile(file, destination, true);
```
Comment (sukamat) [inline actions/fgCopyActionHelper.js]: @raga-adbe-gh This is calling the saveFile() which still has the delete file on locked code as I see in `sharepoint.js`. So for copy from source to pink, it is still overwriting but for promote it is creating a new version?

Comment (raga-adbe-gh) [reply actions/fgCopyActionHelper.js]: Thanks @sukamat, Correct for copy flow is unchanged. The content write (new version) is only added for promote when content is moved from pink tree to main tree.

**Review by arshadparwaiz — APPROVED** [review]

**Comment by sukamat** [comment]
@raga-adbe-gh As discussed this morning, let's update the Copy to use the same upload API that we are using for Promote. The session creation functionality is no longer required I think, so it can be cleaned up or can be left as-is. No strong preference on that.

**Comment by raga-adbe-gh** [comment]
> @raga-adbe-gh As discussed this morning, let's update the Copy to use the same upload API that we are using for Promote. The session creation functionality is no longer required I think, so it can be cleaned up or can be left as-is. No strong preference on that.

Thanks @sukamat , As suggested updated the flow

**Review by sukamat — APPROVED** [review]

**Pull request merged**